src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/LanguageServerEndpointAttribute.cs

```diff
@@ -17,8 +17,14 @@ public class LanguageServerEndpointAttribute : Attribute
     /// </summary>
     public string Method { get; }
 
-    public LanguageServerEndpointAttribute(string method)
+    /// <summary>
+    /// Contains an optional language name that this <see cref="IMethodHandler"/> is specific to.
+    /// </summary>
+    public string? Language { get; }
+
+    public LanguageServerEndpointAttribute(string method, string? language = null)
```

Comment:
i'd prefer thsi be non-optional.  usages of this should be explicit if they're for all languages or not.

Reply:
How should we describe that?  Null seems OK as a way to indicate that it is for 'any'

Reply:
Going through this some more -  I think I agree with Cyrus.  We should just be explicit and allow a list of languages here.

The MethodAttribute we can then define to be C#/VB/F# (for all the C# ones) so we don't have to update literally every handler itself.  In separate PR we can rename and extract it out.

Reply:
Switched to params string[] languages. This way other implementations are still able to not use languages, but our MethodAttribute now specifies C#, F# and VB as languages.